utils/url/url.go

```diff
@@ -0,0 +1,136 @@
+package url
+
+import (
+	"strings"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/reflection"
+)
+
+const defaultPathSeparator = "/"
+
+// IsParamSegment checks whether the segment string is a path parameter
+func IsParamSegment(segment string) bool {
```

Comment:
mention OpenAPI because the {} is only defined by this spec https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#path-templating